src/x86/rfxcodec_encode_dwt_shift_x86_sse2.asm

```diff
@@ -21,6 +22,29 @@
 
 %ifidn __OUTPUT_FORMAT__,elf
 section .note.GNU-stack noalloc noexec nowrite progbits
+%ifdef PIC
+section .text
+extern _GLOBAL_OFFSET_TABLE_
+..@get_GOT:
+	mov ebx,[esp]
+	ret
+%define lsym(name) ebx + name wrt ..gotoff
+%macro get_GOT 0
+	call ..@get_GOT
+	add ebx,_GLOBAL_OFFSET_TABLE_+$$-..@get_GOT wrt ..gotpc
+%endmacro
+%else
+%endif
+%else
+; not ELF
+%ifdef PIC
+%error Position-Independent Code is currently only supported for ELF
```

Comment:
Yes, I have evidence. Proof, actually. This is the way DLLs are implemented on ELF systems.

NASM actually does support BSD-style a.out DLLs with the same syntax, but given that they were only used on versions of FreeBSD, MirBSD, NetBSD and OpenBSD over ten years old, it’s probably safe to err on the safe side (especially as all other a.out systems don’t support that).

The C code can probably utilise something like `#if !defined(PIC) || defined(__ELF__)` around the places where it sets the asm variants. Do you wish me to add that to the pull request?